### PR TITLE
Whole field truncation.

### PR DESCRIPTION
**Pull request by matthewperry**

### CODE REVIEW

File: includes/config.inc [hunk 1]
```diff
@@ -494,34 +494,79 @@ function islandora_solr_metadata_config_field_form($form, &$form_state, $config_
...
     ),
     'word_safe' => array(
       '#type' => 'checkbox',
       '#title' => t('Word-safe'),
       '#description' => t('Attempt to truncate on a word boundary. See truncate_utf8() for more info.'),
       '#default_value' => $get_default(array('truncation', 'word_safe'), FALSE),
+      '#states' => array(
+        'visible' => array(
+          ':input[name="wrapper[truncation][truncation_type]"]' => array(
+            'value' => 'separate_value_option',
```
Comment (willtp87) [inline includes/config.inc]: I see no reason not to have these options for both truncation types.

File: theme/theme.inc [hunk 2]
```diff
@@ -229,6 +229,53 @@ function islandora_solr_metadata_query_fields($object, &$solr_fields) {
...
+              $character_output_count += strlen($current_value);
+            }
+
+            $updated_value[]
+              = '<span class="toggle-wrapper">' .
+              t("<span>!value \n<a href='#' class='toggler'>Show more</a></span>", array(
+                '!value' => implode("\n", $list),
+              )) .
+              t("<span>!original_value \n<a href='#' class='toggler'>Show less</a></span>", array(
+                '!original_value' => implode("\n", $list_full),
```
Comment (adam-vessey) [inline theme/theme.inc]: ~~`$list` and `$list_full` might not exist if there were somehow no values for the field?~~ Derp, this is protected against by https://github.com/Islandora/islandora_solr_metadata/pull/26/files#diff-8964427b21f6180b29573c0367f4b3e6R236

Also, every later truncated field will contain everything from previous truncated fields?

Comment (adam-vessey) [inline theme/theme.inc]: (`$list` and `$list_full` are never initially assigned as empty arrays, is what I'm getting at)

File: theme/theme.inc [hunk 3]
```diff
@@ -229,6 +229,53 @@ function islandora_solr_metadata_query_fields($object, &$solr_fields) {
...
+            drupal_add_js("$mod_path/js/truncation-toggle.js");
+
+            // Build two arrays for display, one filtered by size (based on
+            // character count of what is displayed in the field) and one with
+            // all the data.
+            $character_output_count = 0;
+            $truncated = FALSE;
+            foreach ($field['value'] as $index => $current_value) {
+              $list_full[] = $current_value;
+              if ($character_output_count + strlen($current_value) <= $truncation['max_length']) {
```
Comment (adam-vessey) [inline theme/theme.inc]: Same business with the `strlen()` calls here as in https://github.com/Islandora/islandora_solr_search/pull/233

File: theme/theme.inc [hunk 4]
```diff
@@ -169,7 +169,7 @@ function islandora_solr_metadata_query_fields($object, &$solr_fields) {
             'hyperlink' => 0,
             'formatter' => NULL,
           );
-          if (($truncation = drupal_array_get_nested_value($field_config, array('truncation'))) && $truncation['max_length'] > 0) {
+          if (($truncation = drupal_array_get_nested_value($field_config, array('truncation'))) && $truncation['max_length'] > 0 && (!isset($truncation['truncation_type']) || $truncation['truncation_type'] == 'separate_value_option')) {
             $value = truncate_utf8($value, $truncation['max_length'], $truncation['word_safe'], $truncation['ellipsis'], $truncation['min_wordsafe_length']);
             if ($value !== $original_value) {
               $mod_path = drupal_get_path('module', 'islandora_solr_metadata');
```
Comment (adam-vessey) [inline theme/theme.inc]: Just to note: Will have to change when moving the JS for https://github.com/Islandora/islandora_solr_search/pull/233/files#r35256270

**Comment by adam-vessey** [comment]
Just to make it explicit, this is dependent on https://github.com/Islandora/islandora_solr_search/pull/233

**Pull request merged**